controller/telemetry/server.go

```diff
@@ -282,15 +292,15 @@ func (s *server) Report(ctx context.Context, req *write.ReportRequest) (*write.R
 				return nil, errors.New("ResponseCtx is required")
 			}
 
-			for _, latency := range responseScope.ResponseLatencies {
```

Comment:
Since we're now relying on pre-known buckets, we should assert or check that this array has length exactly `len(latencyBucketMaxValues)`

Reply:
Okay, what is the preferred way to "assert or check" in the controller, and what should happen if the array is malformed?